### PR TITLE
[WIP] First official version : 0.1.0

### PR DESCRIPTION
**Pull request by miniche**
Working on the first official version : 0.1.0 (alpha).

Remains to be done : 
- [ ] More tests on more OS version (Ubuntu, Debian, Raspian, Mac)
- [ ] Write documentation
- [ ] And the ultimate Ninir review :smile:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,4 +48,10 @@ nbproject/*
 
 # JetBrains #
 #############
-.idea/*
\ No newline at end of file
+.idea/*
+*/.idea/*
```
Comment (Ninir) [inline .gitignore:52]: You should add these rules to a global gitignore in order not to duplicate this file in every project you have.
https://help.github.com/articles/ignoring-files

File: README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# CentralReport
+
+**Central Report is designed to monitor very easily computers and servers, runs on many current operating systems.**
```
Comment (Ninir) [inline README.md]: Typo: `[...] and runs on many current operating systems.`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Right now, CentralReport can monitor :
```
Comment (Ninir) [inline README.md]: `Right now, Central Report monitors:`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## The idea
+
+We have many ideas to simplify server monitoring:
+
+* Very simple installation (one line in a terminal)
+* Cross platform (Unix, Linux, Windows)
+* Standalone or Cloud conception
+* Alerts (mobile push, mail, etc…)
+* High customizable dashboard
```
Comment (Ninir) [inline README.md]: Typo: Highly

File: README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+## Install or Uninstall CentralReport
+Please visit [Wiki installation page](https://github.com/miniche/CentralReport/wiki/Installation) to get the procedure depending on your system.
+
+
+## Current status
+**The first public alpha version is very close!**
+
+CentralReport runs on **Debian, Ubuntu and Mac OS X**.
+Right now, it's just a standalone version. It's free and very easy : install CentralReport and active the small web server included. You can monitor your host with a simple web browser :
```
Comment (Ninir) [inline README.md]: `[...] It's free and very easy to use:`

File: README.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+**The first public alpha version is very close!**
+
+CentralReport runs on **Debian, Ubuntu and Mac OS X**.
+Right now, it's just a standalone version. It's free and very easy : install CentralReport and active the small web server included. You can monitor your host with a simple web browser : 
+
+![Mac Screenshot](https://raw.github.com/miniche/CentralReport/develop/tools/screenshots/Capture_Mac.png)
+
+
+
+__We are working on the cloud solution : you will be able to view all yours computers and servers, and set some email or push alerts.__
```
Comment (Ninir) [inline README.md]: Typo: on `a` cloud solution

File: README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+
+## Get in touch
+We are open to the world. If you want others informations, or if you want to help us, you can contact us.
```
Comment (Ninir) [inline README.md]: Typo: [...]. If you want more information or you are in a mood to collaborate, feel free to contact us.

File: bash/debian.inc.sh [hunk 8]
```diff
@@ -0,0 +1,353 @@
+#!/bin/bash
+
+# CentralReport Unix/Linux Indev version.
+# Be careful! Don't use in production environment!
+
+# This file contain all functions to manage CR install/unistall on a Debian/Ubuntu distribution.
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: This files containS

File: bash/debian.inc.sh [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+        return 0
+    fi
+}
+
+function debian_remove_config {
+
+    logFile "Removing CentralReport config file..."
+
+    if [ -f ${CONFIG_FILE} ]; then
+        displayAndExec "Remove existing config file..." rm -fv $CONFIG_FILE
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: Removing existing

File: bash/debian.inc.sh [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error when deleting CentralReport config file at ${CONFIG_FILE} (Error code : ${RETURN_CODE})"
+            return ${RETURN_CODE}
+        else
+            logFile "CentralReport config file deleted"
```
Comment (Ninir) [inline bash/debian.inc.sh:110]: You should either add a point or an exclamation mark

File: bash/debian.inc.sh [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+        return 0
+    fi
+}
+
+function debian_remove_startup_script {
+
+    logFile "Removing startup script..."
+
+    if [ -f ${STARTUP_DEBIAN} ]; then
+        displayAndExec "Remove existing startup script..." rm -rfv $STARTUP_DEBIAN
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: Removing

File: bash/debian.inc.sh [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+        else
+
+            displayAndExec "Removing startup service..." update-rc.d -f centralreport.sh remove
+            RETURN_CODE="$?"
+
+            if [ ${RETURN_CODE} -ne "0" ]; then
+                logError "Error when removing startup script with update-rc.d (Error code : ${RETURN_CODE})"
+                return ${RETURN_CODE}
+            else
+                logFile "Startup script deleted"
```
Comment (Ninir) [inline bash/debian.inc.sh:139]: You should either add a point or an exclamation mark

File: bash/debian.inc.sh [hunk 13]
```diff
@@ -0,0 +1,353 @@
...
+# --
+function debian_cp_bin {
+    # Copy CentralReport files in the right directory.
+    mkdir ${INSTALL_DIR}
+
+    if [ $? -ne "0" ]; then
+          logError "Error when creating CentralReport dir at ${INSTALL_DIR} (Error code : $?)"
+          return 1
+    else
+        displayAndExec "Copy CentralReport in the good directory..." cp -R -f -v centralreport ${PARENT_DIR}
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: Copying

File: bash/debian.inc.sh [hunk 14]
```diff
@@ -0,0 +1,353 @@
...
+        else
+            return 0
+        fi
+    fi
+}
+
+function debian_cp_startup_plist {
+    # Copy startup plist for launchd in the right directory.
+
+    displayAndExec "Copy startup script in the good directory..." cp -f -v ${STARTUP_DEBIAN_INSTALL} ${STARTUP_DEBIAN}
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: Copying

Comment (Ninir) [inline bash/debian.inc.sh]: Instead of saying this, what do you think of showing: `Installing the startup script...`?

File: README.md (hunk 2 above)
Comment (Guimove) [inline README.md]: Central Report is designed to easily monitor computers and servers [...]  ?

File: README.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
+# CentralReport
+
+**Central Report is designed to monitor very easily computers and servers, runs on many current operating systems.**
+
+Right now, CentralReport can monitor :
+
+* CPU
+* Physical Memory (RAM)
+* Swap
```
Comment (Guimove) [inline README.md]: Virtual Memory (Swap)

File: bash/debian.inc.sh [hunk 16]
```diff
@@ -0,0 +1,353 @@
...
+        fi
+    fi
+}
+
+# --
+# Install procedure
+# --
+function debian_install {
+
+    # Install only can perform if current user is root (or have administrative privileges)
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: `Installation can only be performed if the current user has administrative privileges.`

File: bash/debian.inc.sh [hunk 17]
```diff
@@ -0,0 +1,353 @@
...
+
+    if [ $? -ne "0" ]; then
+          logError "Error when creating CentralReport dir at ${INSTALL_DIR} (Error code : $?)"
+          return 1
+    else
+        displayAndExec "Copy CentralReport in the good directory..." cp -R -f -v centralreport ${PARENT_DIR}
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error when copying CentralReport bin files in ${PARENT_DIR} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/debian.inc.sh]: Error copying [...]

File: bash/debian.inc.sh [hunk 18]
```diff
@@ -0,0 +1,353 @@
...
+      logError "Error when copying startup script at ${STARTUP_PLIST} (Error code : ${RETURN_CODE})"
+      return ${RETURN_CODE}
+    else
+        chmod 755 ${STARTUP_DEBIAN}
+
+        displayAndExec "Registering startup script" update-rc.d centralreport.sh defaults
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ]; then
+          logError "Error when registering startup script with update-rc.d (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/debian.inc.sh]: Error registering [...]

File: bash/debian.inc.sh [hunk 19]
```diff
@@ -0,0 +1,353 @@
...
+}
+
+function debian_cp_startup_plist {
+    # Copy startup plist for launchd in the right directory.
+
+    displayAndExec "Copy startup script in the good directory..." cp -f -v ${STARTUP_DEBIAN_INSTALL} ${STARTUP_DEBIAN}
+    RETURN_CODE="$?"
+
+    if [ ${RETURN_CODE} -ne "0" ]; then
+      logError "Error when copying startup script at ${STARTUP_PLIST} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/debian.inc.sh]: Remove "when"

File: utils/launchers/mac/com.centralreport.plist [hunk 20]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<plist version="1.0">
+<dict>
```
Comment (Ninir) [inline utils/launchers/mac/com.centralreport.plist:4]: Please add one level of indentation

File: utils/launchers/debian/centralreport.sh [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+# Default-Stop:      0 1 6
+# Short-Description: Start CentralReport at boot time
+# Description:       Enable service provided by daemon.
+### END INIT INFO
+
+
+# Carry out specific functions when asked to by the system
+case "$1" in
+  start)
+    echo "Starting CentralReport "
```
Comment (Ninir) [inline utils/launchers/debian/centralreport.sh]: Please remove the additional space before `"`

File: uninstall.sh [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+            logInfo "Thanks!"
+            logConsole "\033[0m"
+
+        fi
+    fi
+fi
+
+# End of program
+logConsole " "
+logInfo " -- End of program -- "
```
Comment (Ninir) [inline uninstall.sh]: Typo: `End of the program`

File: bash/functions.inc.sh [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+
+# Displays the message with current status (.../ERR/OK), while executing the command.
+# First parameter: MESSAGE
+# Others parameters: COMMAND (! not |)
+function displayAndExec() {
+    local message=$1
+    echo -n "[...] ${message}"
+    shift
+
+    echo ">>> $*" >> /dev/null 2>&1
```
Comment (Guimove) [inline bash/functions.inc.sh:49]: You should add a log file : 

```
DATE=`date +"%Y%m%d%H%M%S"`
LOG_FILE="/tmp/CentralReport-$DATE.log"
```

and replace 
    /dev/null 2>&1 
by 
    $LOG_FILE 2>&1

File: bash/functions.inc.sh [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+    sh -c "$*" >> /dev/null 2>&1
```
Comment (Guimove) [inline bash/functions.inc.sh:50]: Same here for log file

File: bash/macos.inc.sh [hunk 25]
```diff
@@ -0,0 +1,359 @@
...
+
+    if [ ! -f ${PID_FILE} ]; then
+        logInfo "CentralReport is already stopped!"
+        return 0
+    else
+        sudo python ${INSTALL_DIR}/centralreport.py stop
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ] && [ ${RETURN_CODE} -ne "143" ]; then
+            logError "Error when stopping CentralReport (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/macos.inc.sh]: Remove when

File: bash/macos.inc.sh [hunk 26]
```diff
@@ -0,0 +1,359 @@
...
+            return ${RETURN_CODE}
+        else
+            logInfo "CentralReport stopped"
```
Comment (Guimove) [inline bash/macos.inc.sh:52]: You should add a !

File: bash/macos.inc.sh [hunk 27]
```diff
@@ -0,0 +1,359 @@
...
+function macos_remove_bin {
+
+    logFile "Removing CentralReport bin files..."
+
+    if [ -d ${INSTALL_DIR} ]; then
+        displayAndExec "Remove existing install directory..." sudo rm -rfv ${INSTALL_DIR}
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error when deleting CentralReport bin directory at ${INSTALL_DIR} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/macos.inc.sh]: Remove when

File: bash/macos.inc.sh [hunk 28]
```diff
@@ -0,0 +1,359 @@
...
+        return 0
+    fi
+}
+
+function macos_remove_config {
+
+    logFile "Removing CentralReport config file..."
+
+    if [ -f ${CONFIG_FILE} ]; then
+        displayAndExec "Remove existing config file..." sudo rm -fv ${CONFIG_FILE}
```
Comment (Guimove) [inline bash/macos.inc.sh]: Removing...

File: bash/macos.inc.sh [hunk 29]
```diff
@@ -0,0 +1,359 @@
...
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error when deleting CentralReport config file at ${CONFIG_FILE} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/macos.inc.sh]: Remove when

File: bash/macos.inc.sh [hunk 30]
```diff
@@ -0,0 +1,359 @@
...
+            return ${RETURN_CODE}
+        else
+            logFile "CentralReport config file removed"
```
Comment (Guimove) [inline bash/macos.inc.sh:114]: You should had a !

File: bash/macos.inc.sh [hunk 31]
```diff
@@ -0,0 +1,359 @@
...
+        return 0
+    fi
+}
+
+function macos_remove_startup_plist {
+
+    logFile "Removing startup plist..."
+
+    if [ -f ${STARTUP_PLIST} ]; then
+        displayAndExec "Remove existing startup plist file..." sudo rm -fv ${STARTUP_PLIST}
```
Comment (Guimove) [inline bash/macos.inc.sh]: Removing...

File: bash/macos.inc.sh [hunk 32]
```diff
@@ -0,0 +1,359 @@
...
+        RETURN_CODE="$?"
+
+        if [ $? -ne "0" ]; then
+            logError "Error when deleting startup plist file at ${STARTUP_PLIST} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/macos.inc.sh]: Remove when

File: bash/macos.inc.sh [hunk 33]
```diff
@@ -0,0 +1,359 @@
...
+            return ${RETURN_CODE}
+        else
+            logFile "Startup plist removed"
```
Comment (Guimove) [inline bash/macos.inc.sh:135]: You should add a !

File: bash/macos.inc.sh [hunk 34]
```diff
@@ -0,0 +1,359 @@
...
+    fi
+    if [ ! -d "/usr/local/bin" ]; then
+        sudo mkdir /usr/local/bin
+    fi
+
+    sudo mkdir ${INSTALL_DIR}
+    RETURN_CODE="$?"
+
+    if [ ${RETURN_CODE} -ne "0" ]; then
+          displayError "Error when creating CentralReport dir at ${INSTALL_DIR} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/macos.inc.sh]: Remove when

File: bash/macos.inc.sh [hunk 35]
```diff
@@ -0,0 +1,359 @@
...
+          return ${RETURN_CODE}
+    else
+        displayAndExec "Copying CentralReport in the good directory..." sudo cp -R -f -v centralreport ${PARENT_DIR}
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            displayError "Error when copying CentralReport bin files in ${PARENT_DIR} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/macos.inc.sh]: Remove when

File: bash/macos.inc.sh [hunk 36]
```diff
@@ -0,0 +1,359 @@
...
+}
+
+function macos_cp_startup_plist {
+    # Copy startup plist for launchd in the right directory.
+
+    displayAndExec "Copying startup plist in the good directory..." sudo cp -f -v ${STARTUP_PLIST_INSTALL} ${STARTUP_PLIST}
+    RETURN_CODE="$?"
+
+    if [ ${RETURN_CODE} -ne "0" ]; then
+      displayError "Error when copying startup plist at ${STARTUP_PLIST} (Error code : ${RETURN_CODE})"
```
Comment (Guimove) [inline bash/macos.inc.sh]: F****ing when

Comment (miniche) [inline bash/macos.inc.sh]: Say this to Ninir :P

Comment (Ninir) [inline bash/macos.inc.sh]: You CAN use one of this sentences:
There was an error verb-ing...
There was an error while verb-ing...
There was an error when verb-ing...

Gauthier Wallet
Envoyé depuis mon iPhone.

Comment (Guimove) [inline bash/macos.inc.sh]: Ok 
IMO, in my sys admin experience: 

```
There was an error verb-ing... 
```

Is the sentence structure most used. 
Remember, a good admin/dev is a lazy boy. :walking:

File: bash/debian.inc.sh [hunk 37]
```diff
@@ -0,0 +1,353 @@
...
+        return ${RETURN_CODE}
+    fi
+
+    printTitle "Installing third-party softwares..."
+    logInfo " (Please consult http://github.com/miniche/CentralReport for licenses) "
+
+
+    # Setuptools (easy_install included)
+    printTitle "Installing Setuptools..."
+    displayAndExec "Untar Setuptools..." tar -xzvf ${SETUPTOOLS_TAR} -C thirdparties/
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: `Untaring`

File: bash/debian.inc.sh [hunk 38]
```diff
@@ -0,0 +1,353 @@
...
+
+    cd ${SETUPTOOLS_DIR};
+    displayAndExec "Installing Setuptools..." python setup.py install
+    RETURN_CODE="$?"
+    cd ../../;
+    if [ ${RETURN_CODE} -ne 0 ]; then
+        return ${RETURN_CODE}
+    fi
+
+    displayAndExec "Deleting install files..." rm -Rf ${SETUPTOOLS_DIR}
```
Comment (Ninir) [inline bash/debian.inc.sh]: Typo: `Deleting installation files...`

File: bash/functions.inc.sh [hunk 39]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# CentralReport Unix/Linux Indev version.
+# Be careful! Don't use in production environment!
+
+# Gets current OS (Linux distrib or Unix OS)
+function getOS(){
+
+    if [ $(uname -s) == "Darwin" ]; then
```
Comment (Ninir) [inline bash/functions.inc.sh]: `if  "Darwin" == $(uname -s)`

File: bash/functions.inc.sh [hunk 40]
```diff
@@ -0,0 +1,75 @@
...
+        CURRENT_OS=${OS_MAC}
+    elif [ -f "/etc/debian_version" ] || [ -f "/etc/lsb-release" ]; then
+        CURRENT_OS=${OS_DEBIAN}
+    fi
+}
+
+# Displays the python version (if python is available)
+# Return 0 is python is available, or 1 if an error occured
```
Comment (Ninir) [inline bash/functions.inc.sh]: Typo: `Returns 0 if python is available, 1 otherwise.`

File: utils/launchers/debian/centralreport.sh [hunk 41]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+### BEGIN INIT INFO
+# Provides:          CentralReport
+# Required-Start:    $all
+# Required-Stop:     $all
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Start CentralReport at boot time
+# Description:       Enable service provided by daemon.
```
Comment (Guimove) [inline utils/launchers/debian/centralreport.sh]: Daemon.py or Daemon

File: uninstall.sh [hunk 42]
```diff
@@ -0,0 +1,101 @@
...
+            logError "CentralReport may still be installed on this host"
+
+        else
+            # Ok, it's done !
+            logConsole "\033[1;32m"
+            logConsole " "
+            logInfo "CentralReport might be deleted on your host."
+            logInfo "It's sad, but you're welcome ! :-)"
+            logConsole " "
+            logInfo "PS : You can write to developers if you found bad things in CentralReport."
```
Comment (Guimove) [inline uninstall.sh]: Thanks for your interest in CentralReport ! One of the best ways you can help us improve CentralReport is to let us know about any problems you find with it.

File: bash/macos.inc.sh [hunk 43]
```diff
@@ -0,0 +1,359 @@
...
+
+
+
+# --
+# CentralReport config assistant
+# --
+function macos_config_assistant {
+
+    logConsole "\033[1;32m"
+    logInfo "Lauching CentralReport configuration wizard..."
```
Comment (Ninir) [inline bash/macos.inc.sh]: Typo: `Launching`

File: bash/macos.inc.sh [hunk 44]
```diff
@@ -0,0 +1,359 @@
...
+# --
+# Uninstall functions
+# --
+
+function macos_remove_bin {
+
+    logFile "Removing CentralReport bin files..."
+
+    if [ -d ${INSTALL_DIR} ]; then
+        displayAndExec "Remove existing install directory..." sudo rm -rfv ${INSTALL_DIR}
```
Comment (Ninir) [inline bash/macos.inc.sh]: `installation`

File: bash/macos.inc.sh [hunk 45]
```diff
@@ -0,0 +1,359 @@
...
+        RETURN_CODE="$?"
+
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error when deleting CentralReport bin directory at ${INSTALL_DIR} (Error code : ${RETURN_CODE})"
+            return ${RETURN_CODE}
+        else
+            logFile "CentralReport bin files removed"
```
Comment (Ninir) [inline bash/macos.inc.sh]: `CentralReport Bin files have been removed`

File: bash/macos.inc.sh [hunk 46]
```diff
@@ -0,0 +1,359 @@
...
+        fi
+    else
+        logInfo "CentralReport bin directory doesn't exist!"
+        return 0
+    fi
+}
+
+function macos_remove_config {
+
+    logFile "Removing CentralReport config file..."
```
Comment (Ninir) [inline bash/macos.inc.sh:104]: Typo: Removing CentralReport configuration file...

File: bash/macos.inc.sh [hunk 47]
```diff
@@ -0,0 +1,359 @@
...
+    else
+        logInfo "Startup plist file not found!"
+        return 0
+    fi
+}
+
+
+
+
+
```
Comment (Ninir) [inline bash/macos.inc.sh]: You don't need all of these blank lines

File: bash/macos.inc.sh [hunk 48]
```diff
@@ -0,0 +1,359 @@
...
+
+
```
Comment (Ninir) [inline bash/macos.inc.sh]: You don't need all of these blank lines

File: bash/macos.inc.sh [hunk 49]
```diff
@@ -0,0 +1,359 @@
...
+    # Delete startup plist file
+    macos_remove_startup_plist
+    RETURN_CODE="$?"
+    if [ ${RETURN_CODE} -ne 0 ]; then
+        return ${RETURN_CODE}
+    fi
+
+
+
```
Comment (Ninir) [inline bash/macos.inc.sh]: Please remove uneeded blank lines

File: bash/macos.inc.sh [hunk 50]
```diff
@@ -0,0 +1,359 @@
...
+    # Deleting sudo privileges for this session...
+    sudo -k
+
+    return 0
+}
+
+
+
+
+
```
Comment (Ninir) [inline bash/macos.inc.sh]: Please remove uneeded blank lines

File: bash/macos.inc.sh [hunk 51]
```diff
@@ -0,0 +1,359 @@
...
+# --
+# Uninstall procedure
+# --
+
+function macos_uninstall {
+
+    echo -e "\n\nPlease use your administrator password to uninstall CentralReport on this Mac."
```
Comment (Ninir) [inline bash/macos.inc.sh:293]: We should be consistent with other places and use either `root privileges`, `administrative rights`, etc.

File: centralreport/centralreport.py [hunk 52]
```diff
@@ -0,0 +1,169 @@
...
+import time
+import os
+from cr.daemon import Daemon
+from cr.tools import Config
+from web.server import WebServer
+
+
+class CentralReport(Daemon):
+    configuration = None  # Configuration object
+    isRunning = True  # Bool : True = daemon is running.
```
Comment (Ninir) [inline centralreport/centralreport.py]: This comment is useless  ^^

File: centralreport/centralreport.py [hunk 53]
```diff
@@ -0,0 +1,169 @@
...
+        else:
+            crLog.writeInfo("Webserver is disabled by configuration file")
+
+        if not isError:
+            crLog.writeInfo("CentralReport started!")
+
+            while CentralReport.isRunning:
+                try:
+
+                    if Config.CR_CONFIG_ENABLE_DEBUG_MODE == False:
```
Comment (Ninir) [inline centralreport/centralreport.py]:
``` python
if not Config.CR_CONFIG_ENABLE_DEBUG_MODE:
```

File: centralreport/centralreport.py [hunk 54]
```diff
@@ -0,0 +1,169 @@
...
+                        # If .pid file is not found, we must stop CR (only in production environement)
```
Comment (Ninir) [inline centralreport/centralreport.py]: Typo: `environment`

File: centralreport/centralreport.py [hunk 55]
```diff
@@ -0,0 +1,169 @@
...
+                    crLog.writeFatal("KeyboardInterrupt exception. Stopping CentralReport...")
+                    CentralReport.isRunning = False
+                    self.stop()
+
+        else:
+            crLog.writeError("Error launching CentralReport!")
+
+    def stop(self):
+        """
+            Called when the scripts will be stopped
```
Comment (Ninir) [inline centralreport/centralreport.py]: This is not the PyDoc of the function ;)

File: centralreport/centralreport.py [hunk 56]
```diff
@@ -0,0 +1,169 @@
...
+        """
+
+        crLog.writeInfo("Stopping CentralReport...")
+        self.isRunning = False
+
+        if None != CentralReport.webserver_thread:
```
Comment (Ninir) [inline centralreport/centralreport.py]: Couldn't we use `if CentralReport.webserver_thread is not None` instead?

File: centralreport/config.py [hunk 57]
```diff
@@ -0,0 +1,109 @@
...
+        print('         CentralReport config editor                 ')
+        print('--------------------------------------------------   ')
+        print('\n')
+        print('You can also edit manually the config file, located at /etc/centralreport.cfg')
+
+        print('Stopping CentralReport...')
+        time.sleep(1)
+
+        daemon = CentralReport(Config.CR_PID_FILE)
+        if 0 == daemon.status():
```
Comment (Ninir) [inline centralreport/config.py]: `if not daemon.status()` should work

File: centralreport/config.py [hunk 58]
```diff
@@ -0,0 +1,109 @@
...
+                daemon.stop()
+            except:
+                print('Error stopping CentralReport daemon...')
+
+        config = Config()  # Getting the actual config
+
+        # Enable or disable internal web server
+        print('\n\n')
+        print('-- Internal web server options --')
+        print('CentralReport have a small internal web server to display checks datas, with a simple web browser.')
```
Comment (Ninir) [inline centralreport/config.py]: Typo: `CentralReport has [...]`

File: centralreport/config.py [hunk 59]
```diff
@@ -0,0 +1,109 @@
...
+                print('We do not understand your answer. Please use "yes" or "no"')
+
+        # If the webserver is enabled, we can ask the default port for it.
+        if bool(Config.getConfigValue('Webserver', 'enable')):
+            print('\n')
+            print('Default port is 8080. You can choose a custom port if you want.')
+
+            validPort = False
+            while(not validPort):
+                resultPort = raw_input('Use this port : ')
```
Comment (Ninir) [inline centralreport/config.py]: In english, there is no punctuation before `!`, `?`, `.`, `;`, `:`

File: centralreport/cr/collectors.py [hunk 60]
```diff
@@ -0,0 +1,458 @@
...
+        memoryCheck.resident = mem_resident
+        memoryCheck.swapSize = mem_total
+        memoryCheck.swapFree = 0
+        memoryCheck.swapUsed = mem_swap
+
+        return memoryCheck
+
+    def get_cpu(self):
+        """
+            Gets actual CPU utilization.
```
Comment (Ninir) [inline centralreport/cr/collectors.py]: `Gets the current CPU usage.`

File: bash/functions.inc.sh [hunk 61]
```diff
@@ -0,0 +1,75 @@
...
+        echo -e "\r\033[0;32m [OK ]\033[0m ${message}"
+    fi
+
+    return ${ret}
+}
+
+
+# Verifies if the answer is "Yes" (y/Y/yes/YES/Yes) or not.
+# PARAMETER : a string
+# RETURN : If true, this function return 0 (no error), else return 1 for any other answer.
```
Comment (Ninir) [inline bash/functions.inc.sh]: Returns 0 if true, 1 otherwise

File: bash/macos.inc.sh [hunk 62]
```diff
@@ -0,0 +1,359 @@
+#!/bin/bash
+
+# CentralReport Unix/Linux Indev version.
+# Be careful! Don't use in production environment!
+
+# This file contain all functions to manage CR install/unistall on a Mac.
```
Comment (Ninir) [inline bash/macos.inc.sh]: Typo: `contains`

File: bash/debian.inc.sh (hunk 10 above)
Comment (miniche) [inline bash/debian.inc.sh:110]: It's just a log line. In system.log or any others log files on your system, all sentences never end with a dot.

File: bash/debian.inc.sh (hunk 14 above)
Comment (miniche) [inline bash/debian.inc.sh]: Installation of the startup script is in two steps : copying it, then registering it. And these steps have different messages (line 177 and 186).

File: bash/debian.inc.sh (hunk 10 above)
Comment (Ninir) [inline bash/debian.inc.sh:110]: Even if it's a log line, we should be consistent for every little things we add to the solution.

Comment (miniche) [inline bash/debian.inc.sh:110]: All right. During installation, I've tried to use only '...' (when an action is performed) or ' ! ' when it's an error or an important message, but never a dot when it's just a status

File: utils/launchers/mac/com.centralreport.plist (hunk 20 above)
Comment (miniche) [inline utils/launchers/mac/com.centralreport.plist:4]: I follow official Apple documentation:
http://developer.apple.com/library/mac/#documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html

File: bash/functions.inc.sh (hunk 23 above)
Comment (miniche) [inline bash/functions.inc.sh:49]: Log file is managed by "log.inc.sh". logFile is called line 54 and 57, but I will refactor this in a future commit :)

File: bash/macos.inc.sh (hunk 26 above)
Comment (miniche) [inline bash/macos.inc.sh:52]: It's not an alert or an error, it's just a regular status

File: bash/macos.inc.sh (hunk 30 above)
Comment (miniche) [inline bash/macos.inc.sh:114]: Same here, it's a regular status, not an alert

File: bash/macos.inc.sh (hunk 51 above)
Comment (miniche) [inline bash/macos.inc.sh:293]: Root account is not enabled by default on OS X : you must use an account with administrative rights to get root privileges (with sudo command).
But on Linux, you must use a root account to perform installation. So, vocabulary can't be the same for the two platforms.

File: centralreport/centralreport.py (hunk 52 above)
Comment (miniche) [inline centralreport/centralreport.py]: It's true :smile:

File: centralreport/centralreport.py (hunk 55 above)
Comment (miniche) [inline centralreport/centralreport.py]: Indeed

File: centralreport/centralreport.py (hunk 56 above)
Comment (miniche) [inline centralreport/centralreport.py]: I will try this for a next commit

File: bash/macos.inc.sh (hunk 51 above)
Comment (Ninir) [inline bash/macos.inc.sh:293]: For sure. the thing is: you do need administrative rights no matter you are on Mac or Linux.

File: centralreport/centralreport.py (hunk 56 above)
Comment (miniche) [inline centralreport/centralreport.py]: Yes, it works :wink:

**Pull request closed without merging**